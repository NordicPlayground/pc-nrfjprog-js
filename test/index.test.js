/* Copyright (c) 2015 - 2019, Nordic Semiconductor ASA
 *
 * All rights reserved.
 *
 * Use in source and binary forms, redistribution in binary form only, with
 * or without modification, are permitted provided that the following conditions
 * are met:
 *
 * 1. Redistributions in binary form, except as embedded into a Nordic
 *    Semiconductor ASA integrated circuit in a product or a software update for
 *    such product, must reproduce the above copyright notice, this list of
 *    conditions and the following disclaimer in the documentation and/or other
 *    materials provided with the distribution.
 *
 * 2. Neither the name of Nordic Semiconductor ASA nor the names of its
 *    contributors may be used to endorse or promote products derived from this
 *    software without specific prior written permission.
 *
 * 3. This software, with or without modification, must only be used with a Nordic
 *    Semiconductor ASA integrated circuit.
 *
 * 4. Any software provided in binary form under this license must not be reverse
 *    engineered, decompiled, modified and/or disassembled.
 *
 * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
 * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
 * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
 * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
 * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
 * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
 * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
 * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
 * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
 * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */

const { debug1 } = require('./debug1.test')
const { debug2 } = require('./debug2.test')
const { generic } = require('./generic.test')
const { raceCondition } = require('./racecondition.test')
const { singleDeviceDesctructive } = require('./single-device-destructive.test')
const { singleDeviceNonDesctructive } = require('./single-device-non-destructive.test')
const { rtt } = require('./rtt.test')
const { rttFailing } = require('./rttfailing.test')
const { modemDfu } = require('./modem-dfu-destructive.test')

describe('Debug1', debug1);
describe('Debug2', debug2);
describe('Generic functionality', generic);
describe('Handles race conditions gracefully', raceCondition);
describe('Single device - destructive', singleDeviceDesctructive);
describe('Single device - non-destructive', singleDeviceNonDesctructive);
describe('RTT', rtt);
describe('RTT without RTT firmware', rttFailing);
describe('Modem DFU', rttFailing);

