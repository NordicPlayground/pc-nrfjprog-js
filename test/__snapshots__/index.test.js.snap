// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic functionality throws when too few parameters are sent in 1`] = `"First argument must be a function"`;

exports[`Generic functionality throws when too many parameters are sent in 1`] = `"Too many parameters. The function get library version does not take 2 parameters."`;

exports[`Generic functionality throws when wrong type of parameters are sent in 1`] = `"First argument must be a function"`;

exports[`Modem DFU can not start RTT fails cleanly 1`] = `
[Error: Error occured when start rtt. Errorcode: CouldNotCallFunction (0x9)
Lowlevel error: Unknown value (ffffff24)
]
`;

exports[`RTT fails cleanly when calling other functions before start fails cleanly when calling read without start 1`] = `
[Error: Error occured when rtt read. Errorcode: CouldNotCallFunction (0x9)
Lowlevel error: Unknown value (ffffff02)
]
`;

exports[`RTT fails cleanly when calling other functions before start fails cleanly when calling write without start 1`] = `
[Error: Error occured when rtt write. Errorcode: CouldNotCallFunction (0x9)
Lowlevel error: Unknown value (ffffff02)
]
`;

exports[`RTT opens and close as expected returns an error when using wrong control block location 1`] = `
[Error: Error occured when start rtt. Errorcode: CouldNotCallFunction (0x9)
Lowlevel error: Unknown value (ffffff24)
]
`;

exports[`RTT opens and close as expected returns an error when wrong serialnumber 1`] = `
[Error: Error occured when start rtt. Errorcode: CouldNotCallFunction (0x9)
Lowlevel error: INVALID_PARAMETER (fffffffd)
]
`;

exports[`RTT reads from device reads startmessage 1`] = `"RTT Loopback Session started, DK connected"`;

exports[`RTT reads from device reads startmessage 2`] = `
Array [
  82,
  84,
  84,
  32,
  76,
  111,
  111,
  112,
  98,
  97,
  99,
  107,
  32,
  83,
  101,
  115,
  115,
  105,
  111,
  110,
  32,
  115,
  116,
  97,
  114,
  116,
  101,
  100,
  44,
  32,
  68,
  75,
  32,
  99,
  111,
  110,
  110,
  101,
  99,
  116,
  101,
  100,
]
`;

exports[`RTT without RTT firmware can not start RTT fails cleanly 1`] = `
[Error: Error occured when start rtt. Errorcode: CouldNotCallFunction (0x9)
Lowlevel error: Unknown value (ffffff24)
]
`;

exports[`Single device - non-destructive throws an error when device do not exist 1`] = `
[Error: Error occured when get device info. Errorcode: CouldNotOpenDevice (0x4)
Lowlevel error: EMULATOR_NOT_CONNECTED (fffffff6)
]
`;
